### ISSUE TITLE
Fail to put UUID

### ISSUE THREAD

**Issue by suquark** [ray-project/ray#816]
When I put UUID, it raises NotImplemented.

```
>>> ray.put(uuid.UUID('053af970-3c27-463b-8131-72bf43bd1c01'))
WARNING: Serializing objects of type <class 'uuid.UUID'> by expanding them as dictionaries of their fields. This behavior may be incorrect in some cases.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/suquark/ray_env/lib/python3.6/site-packages/ray/worker.py", line 1625, in put
    worker.put_object(object_id, value)
  File "/Users/suquark/ray_env/lib/python3.6/site-packages/ray/worker.py", line 335, in put_object
    self.store_and_register(object_id, value)
  File "/Users/suquark/ray_env/lib/python3.6/site-packages/ray/worker.py", line 293, in store_and_register
    ray.numbuf.store_list(object_id.id(), self.plasma_client.conn, [value])
numbuf.error: NotImplemented: long overflow
```
UUID is very basic and common type, so I hope ray can provide a solution.

@mitar

**Comment by mitar** [comment]
Oh, this is bad. We have UUID everywhere in our code. :-(

Does UUID not pickle?

**Comment by mitar** [comment]
```python
> pickle.dumps(uuid.uuid4())
"ccopy_reg\n_reconstructor\np0\n(cuuid\nUUID\np1\nc__builtin__\nobject\np2\nNtp3\nRp4\n(dp5\nS'int'\np6\nL67858771142455733375011540860382513964L\nsb."
```

It seems it works.

**Comment by mitar** [comment]
Also with cloudpickle:

```python
>>> import cloudpickle
>>> import uuid
>>> cloudpickle.dumps(uuid.uuid4())
b'\x80\x02cuuid\nUUID\nq\x00)\x81q\x01}q\x02X\x03\x00\x00\x00intq\x03\x8a\x11\x98y\x9a\x1b\xa1b\xa3\x94kI\xa0\xe2\xe2\x0c+\xa0\x00sb.'
```

**Comment by robertnishihara** [comment]
This seems to be a consequence of another bug #549. Since the ID is stored as an int under the hood

```python
import uuid

x = uuid.UUID('053af970-3c27-463b-8131-72bf43bd1c01')

print(x.__dict__)  # {'int': 6952352408881690590081901386870365185}
```

I think I should have a fix for this, let me see.

**Comment by robertnishihara** [comment]
Should be fixed by #821.

**Comment by mitar** [comment]
Thanks!
